### ISSUE TITLE
Error: Cannot find body-parser

### ISSUE THREAD

**Issue by morfert** [Valian/live_vue#9]
I setup a new project following the installation instructions with some modifications. I am not using tailwind so removed the tailwind files and I am using pnpm instead of npm.
```bash
iex -S mix phx.server
```
Errors out. While trouble shooting I tried.
```bash
cd assets

npm run dev
```
It gave me the following error.

```bash
> dev
> vite --host -l warn

failed to load config from C:\Users\moham\code\ganko\assets\vite.config.js
error when starting dev server:
Error: Cannot find module 'body-parser'
Require stack:
- C:\Users\moham\code\ganko\deps\live_vue\assets\js\live_vue\vitePlugin.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Module.require (node:internal/modules/cjs/loader:1234:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (C:\Users\moham\code\ganko\deps\live_vue\assets\js\live_vue\vitePlugin.js:2:20)
    at Module._compile (node:internal/modules/cjs/loader:1375:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1434:10)
    at Module.load (node:internal/modules/cjs/loader:1206:32)
    at Module._load (node:internal/modules/cjs/loader:1022:12)
    at cjsLoader (node:internal/modules/esm/translators:358:17)
```
Note: I am using Windows 11 with elixir 1.16.3 and erlang/otp 27.

**Comment by Valian** [comment]
Hi @morfert ! It looks like vite cannot load `body-parser` module. It should be installed by pnpm when doing `pnpm i ../deps/live_vue`, since it's in `live_vue` package.json. 

But to be honest I didn't try to install in on Windows, only Mac right now.

Maybe we could hop on a quick call to debug it? You can catch me on discord, my username `valian.ai`

**Comment by Valian** [comment]
I think I've reproduced that issue.

```
cd assets
pnpm install
node -e "import('./vite.config.js')"
```

```
node:internal/modules/cjs/loader:1186
  const err = new Error(message);
              ^

Error: Cannot find module 'body-parser'
Require stack:
- /app/deps/live_vue/assets/js/live_vue/vitePlugin.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1186:15)
    at Module._load (node:internal/modules/cjs/loader:1012:27)
    at Module.require (node:internal/modules/cjs/loader:1271:19)
    at require (node:internal/modules/helpers:123:16)
    at Object.<anonymous> (/app/deps/live_vue/assets/js/live_vue/vitePlugin.js:2:20)
    at Module._compile (node:internal/modules/cjs/loader:1434:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1518:10)
    at Module.load (node:internal/modules/cjs/loader:1249:32)
    at Module._load (node:internal/modules/cjs/loader:1065:12)
    at cjsLoader (node:internal/modules/esm/translators:318:15) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/app/deps/live_vue/assets/js/live_vue/vitePlugin.js' ]
}

Node.js v22.2.0
```

but if I'll do just `pnpm run dev` it works without error. 

```
pnpm run dev

> @ dev /app/assets
> vite --host -l warn
```

I'm almost sure your problem is related to not set NODE_ENV correctly. When running `pnpm run` it should be set automatically, maybe it doesn't happen in your case? 🥲 

To check: 
1. find file `node_modules/.vin/vite`, 
2. find line setting NODE_PATH (in my case it's something like `export NODE_PATH="/app/assets/node_modules/.pnpm/node_modules"`, it will be different on windows)
3. Run in manually
4. Try to start vite server as usual by typing `pnpm run dev`.

Maybe this will help us move forward. 

if not, I can bundle all the dependencies in `/priv` and include in the release, so it wouldn't need to install these.

**Comment by morfert** [comment]
Hey @Valian , I  ran `pnpm install` in the `\deps\live_vue\` directory and then switched all my aliased commands in `mix.exs` from `npm` to `pnpm`, I had forgotten to switch these to `pnpm`. I had also forgotten to change the watcher command in `\config\dev.exs` to `pnpm`.
After doing all this it now works properly. 
I think I may have prematurely made the issue and should have spent more time troubleshooting. Thank you.

**Comment by Valian** [comment]
Great, you're welcome! Were the installation instructions easy to follow?

**Comment by morfert** [comment]
Yeah, the instructions were straightforward. 
If I were to give some feedback than it would be that the folloring script is shell specific and does not work on defaut windows shells.
```bash
mkdir vue

for SOURCE in $(find ../deps/live_vue/assets/copy -type f); do
  DEST=${SOURCE#../deps/live_vue/assets/copy/}

  if [ -e "$DEST" ]; then
    echo "SKIPPED $SOURCE -> $DEST. Please update manually"
  else
    echo "COPIED $SOURCE -> $DEST"
    cp $SOURCE $DEST
  fi
donemkdir vue

for SOURCE in $(find ../deps/live_vue/assets/copy -type f); do
  DEST=${SOURCE#../deps/live_vue/assets/copy/}

  if [ -e "$DEST" ]; then
    echo "SKIPPED $SOURCE -> $DEST. Please update manually"
  else
    echo "COPIED $SOURCE -> $DEST"
    cp $SOURCE $DEST
  fi
done
```
Maybe state that it is a script intended for unix. Despite being a windows user I have bash installed, my default shell is nushell though, so just launched bash and ran the script.
I have used a little bit of [live_svelte](https://github.com/woutdp/live_svelte) before and if I recall correctly it contains a `mix task` to setup the files.
You can see in step 3 of [live_svelte installation](https://github.com/woutdp/live_svelte?tab=readme-ov-file#installation) the `mix live_svelte.setup` `mix task` is used which comes from the [live_svelte package](https://github.com/woutdp/live_svelte).
Maybe the script should be replaced with a `mix task` instead.

**Comment by Valian** [comment]
Good idea! I wanted to avoid mix task since I've seen some people complaining they doesn't really understand what happened when installing LiveSvelte. But since it's very similar to what phoenix generators / auth does maybe I could try to go that way. 

Looking forward to hear your feedback! Development is still quite early. I'm working on a landing page with some examples to close #5 , should be live soon ;) 

<img width="1181" alt="image" src="https://github.com/Valian/live_vue/assets/3935986/aa31c0fc-9777-46fd-8930-fd59f4508ef5">

**Comment by Valian** [comment]
I'm reopening. Somehow got this on my own when using pnpm. 

After some digging it seems to be related to using "file:../deps/live_vue" in package.json https://github.com/pnpm/pnpm/issues/3034 - pnpm won't install dependencies of such packages. It kinda sucks 🥲

But I believe I can simply pre-bundle it. Will fix it soon.

**Comment by jcserracampos** [comment]
I just faced the same problem here. If you need any debugging here, hit me up.

I'm using Docker with the `elixir:1.16-otp-26-alpine` image

**Comment by Valian** [comment]
@jcserracampos I just published version `0.3.5`, where that dependency is no longer needed (I'm parsing JSON by myself). Please check now.
